gateway-sp-comms/src/communicator.rs

```diff
@@ -544,6 +550,36 @@ impl ResponseKindExt for ResponseKind {
             }),
         }
     }
+
+    fn try_into_update_start_ack(self) -> Result<(), BadResponseType> {
```

Comment:
I think I originally drafted these named `as_foo()`, but AFAIK all the `as_*()` methods in the standard library succeed unconditionally. I'm hesitant to go with `matches_XXX` because that sounds like it should return a boolean. Would `expect_XXX` be reasonable, or is that too close to `.expect()` (i.e., one might think it panics)?

Reply:
I think `expect_XXX` is perfect.

Reply:
Changed to `expect_XXX` in 518bff00